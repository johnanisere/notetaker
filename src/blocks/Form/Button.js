import styled from "styled-components";

import Button from "../../elements/Button";

export default styled(Button)`
  background: -webkit-linear-gradient(top right, #33bef6, #26aef4);
  color: #ffffff;
  font-size: 15px;
  margin-top: 15px;
  border-radius: 5px;
  cursor: pointer;
  border-color: #33bef6;
  padding: 10px 25px;
  font-weight: 500;
  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
  transition: all 0.15s ease ease-in-out;
  &:hover {
    transform: translateY(-1px);
    box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1), 0 3px 6px rgba(0, 0, 0, 0.08);
  }
`;
