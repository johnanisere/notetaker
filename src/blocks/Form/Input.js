import styled from "styled-components";

import Input from "../../elements/Input";

export default styled(Input)`
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background: rgba(0, 0, 0, 0.03);
`;
