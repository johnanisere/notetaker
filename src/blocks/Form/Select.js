import styled from "styled-components";

import Select from "../../elements/Select";

export default styled(Select)`
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background-color: rgba(0, 0, 0, 0.03);
  appearance: none;
  border: 1px solid #d4d4d4;
  padding: 12px;
  box-sizing: border-box;
  margin-top: 8px;
  border-radius: 4px;
`;
