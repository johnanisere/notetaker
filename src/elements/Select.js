import styled from "styled-components";

const Select = styled.select`
  width: 100%;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 #d7d7e0;
  border: 1px solid #d4d4d4;
  padding: 12px;
  background-image: url(https://buzzhostingservices.com/images/drop-down-arrow-7.png);
  background-size: auto 25px;
  background-color: white;
  background-position: calc(100% - 10px) center;
  background-repeat: no-repeat;
  outline: none;
  transition: outline 0.2s ease-in-out 0s;
`;

export default Select;
