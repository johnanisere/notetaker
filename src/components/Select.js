import React from "react";
import Form from "../blocks/Form";

const Select = ({ list = [], value, onChange, name }) => (
  <Form.Select onChange={onChange} value={value} name={name}>
    <option value="">Select</option>
    {list.map((value, key) => (
      <option value={value} key={key}>
        {value}
      </option>
    ))}
  </Form.Select>
);

export default Select;
