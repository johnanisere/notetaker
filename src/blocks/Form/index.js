import styled from "styled-components";

import Input from "./Input";
import Select from "./Select";
import Button from "./Button";
import Label from "../../elements/Label";

const Form = styled.form``;

Form.Input = Input;
Form.Label = Label;
Form.Select = Select;
Form.Button = Button;

export default Form;
