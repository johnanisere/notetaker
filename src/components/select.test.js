import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";

import Select from "./Select";
import Form from "../blocks/Form";

Enzyme.configure({ adapter: new Adapter() });

it("InputGroup component exists", () => {
  const wrapper = shallow(<Select />);
  expect(wrapper.exists()).toBe(true);
});

it("InputGroup component renders", () => {
  const wrapper = shallow(<Select />);

  expect(wrapper).toMatchSnapshot();
});

it(`Should contain Select`, () => {
  const props = {
    name: "note",
    list: ["red", "blue", "white"],
    onChange: () => {},
    value: "red"
  };
  const wrapper = shallow(<Select {...props} />);

  expect.assertions(2);
  expect(wrapper.find(Form.Select)).toHaveLength(1);
  expect(wrapper.find("option")).toHaveLength(4);
});
