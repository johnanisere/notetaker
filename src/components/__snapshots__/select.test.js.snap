// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <option
          value=""
        >
          Select
        </option>,
        Array [],
      ],
      "name": undefined,
      "onChange": undefined,
      "value": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select",
          "value": "",
        },
        "ref": null,
        "rendered": "Select",
        "type": "option",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 #d7d7e0;
  border: 1px solid #d4d4d4;
  padding: 12px;
  background-image: url(https://buzzhostingservices.com/images/drop-down-arrow-7.png);
  background-size: auto 25px;
  background-color: white;
  background-position: calc(100% - 10px) center;
  background-repeat: no-repeat;
  outline: none;
  transition: outline 0.2s ease-in-out 0s;
",
          "
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background-color: rgba(0, 0, 0, 0.03);
  appearance: none;
  border: 1px solid #d4d4d4;
  padding: 12px;
  box-sizing: border-box;
  margin-top: 8px;
  border-radius: 4px;
",
        ],
      },
      "displayName": "Styled(styled.select)",
      "foldedComponentIds": Array [
        "sc-htpNat",
      ],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "select",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <option
            value=""
          >
            Select
          </option>,
          Array [],
        ],
        "name": undefined,
        "onChange": undefined,
        "value": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select",
            "value": "",
          },
          "ref": null,
          "rendered": "Select",
          "type": "option",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
  width: 100%;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 #d7d7e0;
  border: 1px solid #d4d4d4;
  padding: 12px;
  background-image: url(https://buzzhostingservices.com/images/drop-down-arrow-7.png);
  background-size: auto 25px;
  background-color: white;
  background-position: calc(100% - 10px) center;
  background-repeat: no-repeat;
  outline: none;
  transition: outline 0.2s ease-in-out 0s;
",
            "
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background-color: rgba(0, 0, 0, 0.03);
  appearance: none;
  border: 1px solid #d4d4d4;
  padding: 12px;
  box-sizing: border-box;
  margin-top: 8px;
  border-radius: 4px;
",
          ],
        },
        "displayName": "Styled(styled.select)",
        "foldedComponentIds": Array [
          "sc-htpNat",
        ],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "select",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
